backend/tournesol/views/preview.py

```diff
@@ -329,6 +337,30 @@ def _draw_logo(self, image: Image, entity: Entity, upscale_ratio: int):
                 dest=tuple(numpy.multiply((16, 24), upscale_ratio)),
             )
 
+    def _draw_duration(self, image: Image, entity: Entity, fnt_config, upscale_ratio: int):
+        """
+        Draw the duration on the preview.
+        """
+
+        duration = entity.metadata.get("duration", 0)
+        minutes, seconds = divmod(duration, 60)
+        hours, minutes = divmod(minutes, 60)
+
+        overlay = Image.new('RGBA', ENTITY_DURATION_WH, COLOR_DURATION_RECTANGLE)
```

Comment:
I find that the grey box is sometimes a bit too large.

I think it should be possible to adjust the width of the overlay dynamically based on the text length (See https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.textlength) ?